daemon/exec.go

```diff
@@ -306,6 +307,32 @@ func (daemon *Daemon) ContainerExecStart(ctx context.Context, name string, stdin
 	return nil
 }
 
+// ContainerExecKill kills an exec instance.
+func (daemon *Daemon) ContainerExecKill(ctx context.Context, name string, signame string) error {
+
+	var err error
+	var sig syscall.Signal
+
+	if signame != "" {
+		if sig, err = signal.ParseSignal(signame); err != nil {
+			return errdefs.InvalidParameter(err)
+		}
+		if !signal.ValidSignalForPlatform(sig) {
+			return errdefs.InvalidParameter(fmt.Errorf("the %s daemon does not support signal %d", runtime.GOOS, sig))
+		}
+	} else {
+		// If no signal is passed, default to SIGKILL
+		sig = signal.SignalMap["KILL"]
+	}
+
+	e, err := daemon.getExecConfig(name)
+	if err != nil {
+		return err
+	}
+
+	return daemon.containerd.SignalProcess(ctx, e.ContainerID, name, int(sig))
```

Comment:
Do we know what (kind of) errors this can return?

Reply:
I tried to trace the possibilities here, and I think we either get a `errdefs.NotFound`, or the result of `errdefs.FromGRPC`. I'm not sure which errors we might end up with via the GRPC calls though.